rest/src/main/java/org/n52/web/ctrl/CategoriesParameterController.java

```diff
@@ -29,12 +29,25 @@
 package org.n52.web.ctrl;
 
 
+import org.n52.io.request.IoParameters;
 import org.n52.io.response.CategoryOutput;
+import org.n52.series.db.DataAccessException;
+import org.n52.series.db.dao.DbQuery;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequestMapping(UrlSettings.COLLECTION_CATEGORIES)
 public class CategoriesParameterController extends ParameterRequestMappingAdapter<CategoryOutput> {
 
+    @Override
+    public String getCollectionPath(String hrefBase) {
+        UrlHelper urlhelper = new UrlHelper();
+        return urlhelper.constructHref(hrefBase, UrlSettings.COLLECTION_CATEGORIES);
+    }
+
+    @Override
+    protected int getElementCount(IoParameters queryMap) throws DataAccessException {
+        return super.getEntityCounter().countCategories(new DbQuery(queryMap));
```

Comment:
don't use the `DbQuery` class from the dao module as it is an implementation of the SPI and specific to database access.